api/src/main/java/org/pmiops/workbench/utils/DaoUtils.java

```diff
@@ -0,0 +1,27 @@
+package org.pmiops.workbench.utils;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class DaoUtils {
+
+  /**
+   * Helper for mapping queries. For example, to count all workspaces by active status.
+   *
+   * @param queryResults - Generally dao.findAll()
+   * @param indexer - method to return a value. Frequently a method reference like
+   *     DbUser::getDisabled
+   * @param <ENTITY_T> - Type of entity
+   * @param <ATTRIBUTE_T> - Type of attribute
+   * @return
+   */
+  public static <ENTITY_T, ATTRIBUTE_T> Map<ATTRIBUTE_T, Long> getAttributeToCountMap(
```

Comment:
This generic parameter naming is controversial. I think traditionally this would just be `<T, U>`, but I thought this was was less scary for a reader.

Reply:
The generic problem is orthogonal to the perf problem.

I'd love to do the right query directly, but I couldn't find a good way to do it in our setup (I.e. to return a map from a repository method). Not sure where I found this trick, but I thought it was OK for small tables for now, as we only do this once a minute (and they can hopefully be cached).

Reply:
Spoke with @gjuggler and @freemabd. It sounds like we need to use a `JdbcTemplate` connection or otherwise break out of our existing pattern in order to do the queries necessary. Which seems odd to me: all I need to do is select a bunch of columns and a count column, then bind the results to some kind of value class.

Proposal for now is to check this in and monitor performance of the database on Test and Prod.  I'm also open to adding feature flags, but that seems like overkill. We could simply disable the cron job if it got noticeably bad.